components/service/glean/src/test/java/mozilla/components/service/glean/GleanTest.kt

```diff
@@ -215,6 +218,50 @@ class GleanTest {
         Glean.initialized = true
     }
 
+    @Test
+    fun `queued recorded metrics correctly record during init`() {
+        val counterMetric = CounterMetricType(
+            disabled = false,
+            category = "telemetry",
+            lifetime = Lifetime.Application,
+            name = "counter_metric",
+            sendInPings = listOf("store1")
+        )
+
+        // First set to a pre-init state
+        Glean.initialized = false
+        Dispatchers.API.queueTasks = true
+        Dispatchers.API.testingMode = false
```

Comment:
Why do we need `testingMode` to be false here? 

**Update** Ok I re-read the code and found out :) Left a comment above!